vscode-extension/src/extension.ts

```diff
@@ -657,28 +658,16 @@ async function checkPython() {
       if (error) {
         console.error("Python was not found, can't install requirements");
         console.error("retrieved python path: " + pythonPath);
+        console.error("error: " + error);
 
         // Guide for installation
-        vscode.window
-          .showErrorMessage(
-            "Python is not installed",
-            ...["Install Python", "Retry"]
-          )
-          .then((selection) => {
-            if (selection === "Install Python") {
-              vscode.env.openExternal(
-                vscode.Uri.parse("https://www.python.org/downloads/")
-              );
-            } else if (selection === "Retry") {
-              vscode.commands.executeCommand(COMMANDS.INIT);
-            }
-          });
+        showGuideForInstallation("Specified Python Interpreter is not valid");
```

Comment:
What does it mean for the interpreter to not be valid? To me, this message doesn't really connect to the python installation guide. My thought process would be: "is the interpreter not valid because it's pointing to incorrect path or is python not installed, or what?"

Reply:
> What does it mean for the interpreter to not be valid

It would mean something more along the lines of "it's pointing to an incorrect path". Since a python interpreter/path will be specified at this point, I don't think its valid to say python is not installed since this code doesn't check for existence of python.

What would you recommend as a better message here?

Reply:
I think the message makes sense but then the guide for python installation does not. Instead, they should probably be informed of how to select or configure a working interpreter

Reply:
I think that makes sense, This diff is just refactoring into into helper method using existing logic.